### DISCUSSION TITLE
Doest allowing phantom references disable Garbage Collection?

### DISCUSSION THREAD

**Discussion by joaoluizn** [soot-oss/soot#1513] [Q&A]
I'm using Soot to analyze some Android Apks but I've noticed a tremendous memory consumption while exploring scenes and transforming. 

Since I use soot within my project and it keeps running after sequential analysis, the memory keeps growing slowly and only ends when I stop the execution.

So one of the options that I thought could be making this happens is the Allow phantom refs. Read about a generic context on Phantom refs and they disable garbage collection. I Saw in the soot documentation that it able us to run without all dependencies on the soot classpath, that is the case sometimes when I haven't all dependencies.

So, does Soot allow phantom refs really disable garbage collection?

**Comment by MarcMil** [comment]
Phantom references in the context of soot and garbage collection really have nothing in common. Note that --allow-phantom-refs does have nothing to do with JVMs phantom references.
If anything, using phantom references/classes instead of their actual implementation **saves** system memory, since the actual implementation code cannot be loaded. Allowing phantom refs means that soot is allowed to construct simple class model representations of classes which were not supplied on the soot classpath or process dir. This can happen when you did not specify the soot classpath carefully enough or when there are classes referenced in parts of the code which actually do not exist for some reason, but does not matter if these code parts with dangling references are not loaded at runtime when running the program.

What you may want to do right after your analysis is ```G.reset()```;
After this, the GC should be able to free most soot objects, in particular those which are representing the applications you analyze.
In case you still have memory problems **after** you run ```G.reset()```, please capture a memory snapshot using a tool such as Eclipse Memory Analyzer (MAT). There should not be any SootClasses in that snapshot. If there are, feel free to file a bug report, in that case, soot has a memory leak.
